src/Sulu/Bundle/MediaBundle/Collection/Manager/CollectionManager.php

```diff
@@ -582,7 +587,46 @@ public function delete($id)
         $collectionTitle = $collectionMeta ? $collectionMeta->getTitle() : null;
         $locale = $collectionMeta ? $collectionMeta->getLocale() : null;
 
+        if (!$forceRemoveChildren) {
+            $user = $this->getCurrentUser();
+
+            if (null !== $user) {
+                $totalChildResourcesWithoutPermissions = $this->collectionRepository->countUnauthorizedChildCollectionsOfRootCollection(
+                    $id,
+                    $user,
+                    $this->permissions[PermissionTypes::DELETE]
+                );
+
+                if ($totalChildResourcesWithoutPermissions) {
+                    $childCollectionResourcesWithoutPermissions = $this->collectionRepository->findUnauthorizedChildCollectionResourcesOfRootCollection(
+                        $id,
+                        $user,
+                        $this->permissions[PermissionTypes::DELETE],
+                        3
+                    );
+
+                    throw new InsufficientChildPermissionsException(
+                        $childCollectionResourcesWithoutPermissions,
+                        $totalChildResourcesWithoutPermissions,
+                        PermissionTypes::DELETE
+                    );
+                }
+            }
```

Comment:
maybe we could move this into a method `checkPermissionForDelete();` to use early returns instead of have this nested if statements e.g.:

```php
            $user = $this->getCurrentUser();

            if (null === $user) {
                 return;
            }
            
            $totalChildResourcesWithoutPermissions = $this->collectionRepository->countUnauthorizedChildCollectionsOfRootCollection(
                $id,
                $user,
                $this->permissions[PermissionTypes::DELETE]
            );

            if ($totalChildResourcesWithoutPermissions) {
                 return;
            }
            
           $childCollectionResourcesWithoutPermissions = $this->collectionRepository->findUnauthorizedChildCollectionResourcesOfRootCollection(
                $id,
                $user,
                $this->permissions[PermissionTypes::DELETE],
                3
           );

           throw new InsufficientChildPermissionsException(
                $childCollectionResourcesWithoutPermissions,
                $totalChildResourcesWithoutPermissions,
                PermissionTypes::DELETE
           );
```